### PR TITLE
Bitmap position

### LINKED ISSUE #15: Noto Color Emoji Font weird offset
When using this bundled font in Fyne, replacing the standard emoji font, the render acts strangely, as if the font was cut off. 

image

I decided to investigate and after analyzing the font in FontLab, I realized the emojis for this font were not SVGs but bitmaps. So I came across this line in the render.go file:

```
case api.GlyphBitmap: 
_ = r.drawBitmap(g, format, img, xPos, yPos)
```

After some tests to understand how yPos works, I changed it to 0 and the problem was completely solved (tested on Windows and Mac). 

image

I don't know if I was just super lucky, but I thought it was worth reporting here.

### PR DESCRIPTION
**Pull request by benoitkugler**
This is an alternative to #16 : using the metric computed in `glyph.YBearing`.

In `drawBitmap`,  (x, y) are the position of the baseline and `glyph.Height` is not really what we want since its also includes the _descent_. Fortunately, `glyph.YBearing` precisely gives the distance between the baseline and the top of the bitmap content.

I think it should also fix #15.

### CODE REVIEW

File: render_test.go [hunk 1]
```diff
@@ -137,3 +140,22 @@ func (ff fixedFontmap) ResolveFace(r rune) font.Face {
...
+	text := "\U0001F615\U0001F618\U0001F616"
+	r := &render.Renderer{
+		FontSize: 40,
+		Color:    color.Black,
+	}
+
+	img := image.NewNRGBA(image.Rect(0, 0, 150, 100))
+	draw.Draw(img, img.Bounds(), &image.Uniform{color.White}, image.Point{}, draw.Src)
+	data, _ := ot.Files.ReadFile("bitmap/NotoColorEmoji.ttf")
+	face, _ := font.ParseTTF(bytes.NewReader(data))
```
Comment (whereswaldon) [inline render_test.go:154]: This rasterizes, but doesn't test for correctness. Shouldn't we compare it against the checked-in reference image to ensure we don't break the alignment in the future?

Comment (benoitkugler) [reply render_test.go:154]: Nice suggestion, done !

**Review by andydotxyz — APPROVED** [review]
Fantastic, much better solution thanks.

File: bitmap.go [hunk 2]
```diff
@@ -53,15 +53,15 @@ func (r *Renderer) drawBitmap(g shaping.Glyph, bitmap api.GlyphBitmap, img draw.
...
+		rect := image.Rect(int(x), int(y-fixed266ToFloat(g.YBearing)*r.PixScale), int(x+fixed266ToFloat(g.Width)*r.PixScale), int(y))
 		scale.NearestNeighbor.Scale(img, rect, sub, sub.Bounds(), draw.Over, nil)
 	case api.JPG, api.PNG, api.TIFF:
 		pix, _, err := image.Decode(bytes.NewReader(bitmap.Data))
 		if err != nil {
 			return err
 		}
 
-		rect := image.Rect(int(x), int(y), int(x+fixed266ToFloat(g.Width)*r.PixScale), int(y-fixed266ToFloat(g.Height)*r.PixScale))
+		rect := image.Rect(int(x), int(y-fixed266ToFloat(g.YBearing)*r.PixScale), int(x+fixed266ToFloat(g.Width)*r.PixScale), int(y))
```
Comment (andydotxyz) [inline bitmap.go]: The aspect ratio looks wrong - should `g.Width` be `g.XBearing` so we are using the same coordinate on both directions?

**Review by andydotxyz — CHANGES_REQUESTED** [review]

**Comment by andydotxyz** [comment]
More testing showed a flaw, the faces should be circle

File: bitmap.go (hunk 2 above)
Comment (andydotxyz) [reply bitmap.go]: Oops, it isn't that simple sorry. But I think it needs to know the ratio of height to YBearing and then apply that to the X axis

**Comment by benoitkugler** [comment]
> More testing showed a flaw, the faces should be circle

Ah nice catch ! From the top, we should go down to the end of the glyph, not just to the baseline. Fixed in  [e83f694](https://github.com/go-text/render/pull/17/commits/e83f694b71574a0f5dc8b8249022d1cad137eb14)

**Review by andydotxyz — APPROVED** [review]
Excellent thanks so much!

**Pull request merged**